core/src/util/mod.rs

```diff
@@ -2,6 +2,10 @@ mod decimal128;
 use bson::doc;
 use constants::SQL_ALL_TABLE_TYPES;
 pub use decimal128::Decimal128Plus;
+#[cfg(target_os = "windows")]
+pub mod dsn;
+#[cfg(target_os = "windows")]
+pub(crate) mod odbcinst;
```

Comment:
Marking these two as windows only for now. When we support macos/linux we'll need to figure out which library to link against for the function we need, and dsn can be agnostic.